src/components/recordedit/recordedit.tsx

```diff
@@ -350,56 +447,258 @@ const RecordeditInner = ({
+
+      if (checkPrefill) {
+        // check prefill object for the columns that are being prefilled to update the new forms since we aren't calling getPrefilledDefaultForeignKeyData()
+        if (prefillObject?.fkColumnNames.indexOf(col.name) !== -1) {
+          tempFormValues[`c_${formNumber}-${col.RID}`] = formValues[`c_${lastFormValue}-${col.RID}`];
+        }
+      }
+
+      // copy the raw data (submitted to ermrestjs)
+      col.foreignKey.colset.columns.forEach((col: any) => {
+        const val = formValues[`c_${lastFormValue}-${col.RID}`];
+        if (val === null || val === undefined) return;
+
+        tempFormValues[`c_${formNumber}-${col.RID}`] = val;
+      });
+    });
+
+    return tempFormValues;
+  }
+
+  // show the prefill bulk foreign key modal if we have a prefill object and bulk foreign key recordset props
+  const openBulkForeignKeyModal = () => {
+    const bulkFKObject = reference.bulkCreateForeignKeyObject;
+    // check for bulkCreateForeignKeyObject being defined since a malformed prefillObject should be ignored
+    //   and the `bulkCreateForeignKeyObject` constructor will handle those malformed cases
+    if (!bulkForeignKeyPickerProps || !bulkFKObject) return;
+
+    let getDisabledTuples;
+    if (bulkFKObject.isUnique) {
+      // set getDisabledTuples again since the selected rows could have changed since the last time the modal was opened
+      // selected rows can be changed by updating a single foreign key input, removing the value, or removing a form entirely
+      getDisabledTuples = disabledTuplesPromise(
+        bulkFKObject.leafColumn.reference.contextualize.compactSelectAssociationLink,
+        bulkFKObject.disabledRowsFilter(),
+        bulkForeignKeySelectedRows
+      );
+    }
+
+    setBulkForeignKeyPickerProps({
+      initialReference: bulkForeignKeyPickerProps.initialReference,
+      initialPageLimit: bulkForeignKeyPickerProps.initialPageLimit,
+      config: bulkForeignKeyPickerProps.config,
+      logInfo: bulkForeignKeyPickerProps.logInfo,
+      parentReference: bulkForeignKeyPickerProps.parentReference,
+      getDisabledTuples
+    });
+
+    setShowBulkForeignKeyModal(true);
+  }
+
+  const onSelectedRowsChanged = (rows: SelectedRow[]) => {
+    // return "false" to disable submit button in modal
+    let numForms = forms.length;
+
+    // if we fill the first form, then reduce our calculation by 1
+    // NOTE: forms.length "should" be 1 at this point before subtracting 1
+    if (selectionsFillFirstForm) numForms--;
+
+    return (rows.length + numForms < RECORDEDIT_MAX_ROWS);
+  }
+
+  // user closes the modal without making any selections
+  const closeBulkForeignKeyCB = () => {
+    // if the page was loaded with a modal showing and it is dismissed, update app state variable and do nothing else
+    if (selectionsFillFirstForm) setSelectionsFillFirstForm(false);
```

Comment:
@RFSH pointed out that there was a bug when:
 1. the recordedit app was loaded
 2. the user closed the first modal
 3. then used "Add more" to add 3 values, resulting in 4 forms displayed
 4. remove form 2
 5. open "Add more" again and the wrong row was still disabled

Reply:
The issue was that when the first modal is shown and closed, the first form is initialized and I didn't initialize `bulkForeignKeySelectedRows` to have a `null` value for the first "selection". The following change to this condition will initialize that "array" with a `null` value as the first element representing a null value in the first form.

```js
if (selectionsFillFirstForm) {
  setSelectionsFillFirstForm(false);
  setBulkForeignKeySelectedRows([null]);
}
```